From b063364e0e1a990c0545d3758b0235e7c9ae1c8e Mon Sep 17 00:00:00 2001
From: Nagaraj S <sivasankaranx.nagaraj@intel.com>
Date: Tue, 26 Jul 2016 11:57:05 -0700
Subject: [PATCH 448/639] kernel: Remove CougarMountain codename references

Remove Cougar Mountain codename references from the kernel source
code.

Change-Id: I62935982ce6db6574882dd67b41d60f89fceee3c
Signed-off-by: Nagaraj S <sivasankaranx.nagaraj@intel.com>
---
 drivers/mmc/host/sdhci-acpi.c | 56 +++++++++++++++++++++----------------------
 drivers/mmc/host/sdhci-pci.c  | 48 +++++++++++++++++++++++++++++++------
 drivers/spi/spi-pxa2xx-pci.c  | 28 +++++++++++-----------
 drivers/spi/spi-pxa2xx.c      |  5 ++--
 drivers/usb/dwc3/dwc3-pci.c   |  6 ++---
 5 files changed, 87 insertions(+), 56 deletions(-)

diff --git a/drivers/mmc/host/sdhci-acpi.c b/drivers/mmc/host/sdhci-acpi.c
index 11b8c38..c460ea7 100644
--- a/drivers/mmc/host/sdhci-acpi.c
+++ b/drivers/mmc/host/sdhci-acpi.c
@@ -110,27 +110,27 @@ static void sdhci_acpi_int_hw_reset(struct sdhci_host *host)
 	usleep_range(300, 1000);
 }
 
-void cgm_write_l(struct sdhci_host *host, u32 val, int reg)
+void puma7_write_l(struct sdhci_host *host, u32 val, int reg)
 {
 	if (!(host->flags & SDHCI_DISABLE_REGISTER_WRITE))
 		writel(val, host->ioaddr + reg);
 }
 
-void cgm_write_w(struct sdhci_host *host, u16 val, int reg)
+void puma7_write_w(struct sdhci_host *host, u16 val, int reg)
 {
 	if (!(host->flags & SDHCI_DISABLE_REGISTER_WRITE))
 		writew(val, host->ioaddr + reg);
 }
 
-void cgm_write_b(struct sdhci_host *host, u8 val, int reg)
+void puma7_write_b(struct sdhci_host *host, u8 val, int reg)
 {
 	if (!(host->flags & SDHCI_DISABLE_REGISTER_WRITE))
 		writeb(val, host->ioaddr + reg);
 }
 
-unsigned int sdhci_acpi_cgm_get_min_clock(struct sdhci_host *host)
+unsigned int sdhci_acpi_puma7_get_min_clock(struct sdhci_host *host)
 {
-	/* CGM host controllers only support down to 200kHz */
+	/* PUMA7 host controllers only support down to 200kHz */
 	return 200000;
 }
 
@@ -144,14 +144,14 @@ static const struct sdhci_ops sdhci_acpi_ops_int = {
 };
 
 #ifdef CONFIG_X86_PUMA7
-static const struct sdhci_ops sdhci_acpi_ops_int_cgm_emmc = {
+static const struct sdhci_ops sdhci_acpi_ops_int_puma7_emmc = {
 	.enable_dma = sdhci_acpi_enable_dma,
 	.hw_reset   = sdhci_acpi_int_hw_reset,
-	.get_min_clock = sdhci_acpi_cgm_get_min_clock,
+	.get_min_clock = sdhci_acpi_puma7_get_min_clock,
 #ifdef CONFIG_HW_MUTEXES
-	.write_l = cgm_write_l,
-	.write_w = cgm_write_w,
-	.write_b = cgm_write_b,
+	.write_l = puma7_write_l,
+	.write_w = puma7_write_w,
+	.write_b = puma7_write_b,
 #endif
 };
 
@@ -160,8 +160,8 @@ static const struct sdhci_acpi_chip sdhci_acpi_chip_int = {
 	.ops = &sdhci_acpi_ops_int,
 };
 
-static const struct sdhci_acpi_chip sdhci_acpi_chip_int_cgm_emmc = {
-	.ops = &sdhci_acpi_ops_int_cgm_emmc,
+static const struct sdhci_acpi_chip sdhci_acpi_chip_int_puma7_emmc = {
+	.ops = &sdhci_acpi_ops_int_puma7_emmc,
 };
 
 /*
@@ -238,8 +238,8 @@ static const struct sdhci_acpi_slot sdhci_acpi_slot_int_emmc = {
 	.remove_slot = sdhci_acpi_remove_slot,
 };
 
-static const struct sdhci_acpi_slot sdhci_acpi_slot_int_cgm_emmc1 = {
-	.chip    = &sdhci_acpi_chip_int_cgm_emmc,
+static const struct sdhci_acpi_slot sdhci_acpi_slot_int_puma7_emmc1 = {
+	.chip    = &sdhci_acpi_chip_int_puma7_emmc,
 	.caps    = MMC_CAP_8_BIT_DATA | MMC_CAP_NONREMOVABLE | MMC_CAP_HW_RESET
 		| MMC_CAP_1_8V_DDR,
 	.caps2   = MMC_CAP2_HC_ERASE_SZ | MMC_CAP2_POLL_R1B_BUSY |
@@ -255,8 +255,8 @@ static const struct sdhci_acpi_slot sdhci_acpi_slot_int_cgm_emmc1 = {
 	.remove_slot = sdhci_acpi_remove_slot,
 };
 
-static const struct sdhci_acpi_slot sdhci_acpi_slot_int_cgm_emmc2 = {
-	.chip    = &sdhci_acpi_chip_int_cgm_emmc,
+static const struct sdhci_acpi_slot sdhci_acpi_slot_int_puma7_emmc2 = {
+	.chip    = &sdhci_acpi_chip_int_puma7_emmc,
 	.caps    = MMC_CAP_8_BIT_DATA | MMC_CAP_NONREMOVABLE | MMC_CAP_HW_RESET
 		| MMC_CAP_1_8V_DDR,
 	.caps2   = MMC_CAP2_HC_ERASE_SZ | MMC_CAP2_POLL_R1B_BUSY |
@@ -306,19 +306,17 @@ struct sdhci_acpi_uid_slot {
 };
 
 static const struct sdhci_acpi_uid_slot sdhci_acpi_uids[] = {
-        { "80860F14" , "1" , &sdhci_acpi_slot_int_emmc },
-        { "80860F14" , "3" , &sdhci_acpi_slot_int_sd   },
-		{ "80860F16" , NULL, &sdhci_acpi_slot_int_sd   },
-        { "80862B94" , "1" , &sdhci_acpi_slot_int_cgm_emmc1 },
-        { "80862B95" , "1" , &sdhci_acpi_slot_int_cgm_emmc2 },
-        { "80862B96" , "1" , &sdhci_acpi_slot_int_sd   },
-        { "INT33BB"  , "2" , &sdhci_acpi_slot_int_sdio },
-        { "INT33BB"  , "3" , &sdhci_acpi_slot_int_sd   },
-        { "INT33C6"  , NULL, &sdhci_acpi_slot_int_sdio },
-        { "INT3436"  , NULL, &sdhci_acpi_slot_int_sdio },
-		{ "INT344D"  , NULL, &sdhci_acpi_slot_int_sdio },
-        { "PNP0D40"  },
-        { },
+	{ "80860F14" , "1" , &sdhci_acpi_slot_int_emmc },
+	{ "80860F14" , "3" , &sdhci_acpi_slot_int_sd   },
+	{ "80862B94" , "1" , &sdhci_acpi_slot_int_puma7_emmc1 },
+	{ "80862B95" , "1" , &sdhci_acpi_slot_int_puma7_emmc2 },
+	{ "80862B96" , "1" , &sdhci_acpi_slot_int_sd   },
+	{ "INT33BB"  , "2" , &sdhci_acpi_slot_int_sdio },
+	{ "INT33BB"  , "3" , &sdhci_acpi_slot_int_sd   },
+	{ "INT33C6"  , NULL, &sdhci_acpi_slot_int_sdio },
+	{ "INT3436"  , NULL, &sdhci_acpi_slot_int_sdio },
+	{ "PNP0D40"  },
+	{ },
 };
 
 static const struct acpi_device_id sdhci_acpi_ids[] = {
diff --git a/drivers/mmc/host/sdhci-pci.c b/drivers/mmc/host/sdhci-pci.c
index 0d87fec..2bd8f30 100644
--- a/drivers/mmc/host/sdhci-pci.c
+++ b/drivers/mmc/host/sdhci-pci.c
@@ -28,6 +28,40 @@
 #include <linux/gpio.h>
 #include <linux/pm_runtime.h>
 #include <linux/mmc/sdhci-pci-data.h>
+#ifdef CONFIG_HW_MUTEXES
+#include <linux/hw_mutex.h>
+#endif
+#include "sdhci.h"
+
+/*
+ * PCI device IDs
+ */
+#define PCI_DEVICE_ID_INTEL_PCH_SDIO0	0x8809
+#define PCI_DEVICE_ID_INTEL_PCH_SDIO1	0x880a
+#define PCI_DEVICE_ID_INTEL_BYT_EMMC	0x0f14
+#define PCI_DEVICE_ID_INTEL_BYT_SDIO	0x0f15
+#define PCI_DEVICE_ID_INTEL_BYT_SD	0x0f16
+#define PCI_DEVICE_ID_INTEL_BYT_EMMC2	0x0f50
+#define PCI_DEVICE_ID_INTEL_PUMA7_EMMC	0x2B94
+#define PCI_DEVICE_ID_INTEL_PUMA7_EMMC2	0x2B95
+#define PCI_DEVICE_ID_INTEL_PUMA7_SD	    0x2B96
+
+/*
+ * PCI registers
+ */
+
+#define PCI_SDHCI_IFPIO			0x00
+#define PCI_SDHCI_IFDMA			0x01
+#define PCI_SDHCI_IFVENDOR		0x02
+
+#define PCI_SLOT_INFO			0x40	/* 8 bits */
+#define  PCI_SLOT_INFO_SLOTS(x)		((x >> 4) & 7)
+#define  PCI_SLOT_INFO_FIRST_BAR_MASK	0x07
+
+#define MAX_SLOTS			8
+
+struct sdhci_pci_chip;
+struct sdhci_pci_slot;
 
 #include "sdhci.h"
 #include "sdhci-pci.h"
@@ -294,7 +328,7 @@ static int byt_sdio_probe_slot(struct sdhci_pci_slot *slot)
 	return 0;
 }
 
-static int cgm_emmc_probe_shared_slot(struct sdhci_pci_slot *slot)
+static int puma7_emmc_probe_shared_slot(struct sdhci_pci_slot *slot)
 {
 	slot->host->mmc->caps |= MMC_CAP_8_BIT_DATA | MMC_CAP_NONREMOVABLE |
 				 MMC_CAP_HW_RESET;
@@ -344,9 +378,9 @@ static const struct sdhci_pci_fixes sdhci_intel_mrfl_mmc = {
 	.probe_slot	= intel_mrfl_mmc_probe_slot,
 };
 
-static const struct sdhci_pci_fixes sdhci_intel_cgm_shared_emmc = {
+static const struct sdhci_pci_fixes sdhci_intel_puma7_shared_emmc = {
 	.allow_runtime_pm = true,
-	.probe_slot	= cgm_emmc_probe_shared_slot,
+	.probe_slot	= puma7_emmc_probe_shared_slot,
 };
 
 /* O2Micro extra registers */
@@ -962,15 +996,15 @@ static const struct pci_device_id pci_ids[] = {
 
 	{
 		.vendor		= PCI_VENDOR_ID_INTEL,
-		.device		= PCI_DEVICE_ID_INTEL_CGM_EMMC,
+		.device		= PCI_DEVICE_ID_INTEL_PUMA7_EMMC,
 		.subvendor	= PCI_ANY_ID,
 		.subdevice	= PCI_ANY_ID,
-		.driver_data	= (kernel_ulong_t)&sdhci_intel_cgm_shared_emmc,
+		.driver_data = (kernel_ulong_t)&sdhci_intel_puma7_shared_emmc,
 	},
 
 	{
 		.vendor		= PCI_VENDOR_ID_INTEL,
-		.device		= PCI_DEVICE_ID_INTEL_CGM_EMMC2,
+		.device		= PCI_DEVICE_ID_INTEL_PUMA7_EMMC2,
 		.subvendor	= PCI_ANY_ID,
 		.subdevice	= PCI_ANY_ID,
 		.driver_data	= (kernel_ulong_t)&sdhci_intel_byt_emmc,
@@ -978,7 +1012,7 @@ static const struct pci_device_id pci_ids[] = {
 
 	{
 		.vendor		= PCI_VENDOR_ID_INTEL,
-		.device		= PCI_DEVICE_ID_INTEL_CGM_SD,
+		.device		= PCI_DEVICE_ID_INTEL_PUMA7_SD,
 		.subvendor	= PCI_ANY_ID,
 		.subdevice	= PCI_ANY_ID,
 		.driver_data	= (kernel_ulong_t)&sdhci_intel_byt_sd,
diff --git a/drivers/spi/spi-pxa2xx-pci.c b/drivers/spi/spi-pxa2xx-pci.c
index 25e9660..900962d 100644
--- a/drivers/spi/spi-pxa2xx-pci.c
+++ b/drivers/spi/spi-pxa2xx-pci.c
@@ -19,8 +19,8 @@ enum {
 	PORT_BSW0,
 	PORT_BSW1,
 	PORT_BSW2,
-	PORT_CGM0,
-	PORT_CGM1,
+	PORT_PUMA7_0,
+	PORT_PUMA7_1,
 };
 
 struct pxa_spi_info {
@@ -44,10 +44,10 @@ static struct dw_dma_slave bsw1_rx_param = { .src_id = 7 };
 static struct dw_dma_slave bsw2_tx_param = { .dst_id = 8 };
 static struct dw_dma_slave bsw2_rx_param = { .src_id = 9 };
 
-static struct dw_dma_slave cgm0_tx_param = { .dst_id = 0 };
-static struct dw_dma_slave cgm0_rx_param = { .src_id = 1 };
-static struct dw_dma_slave cgm1_tx_param = { .dst_id = 6 };
-static struct dw_dma_slave cgm1_rx_param = { .src_id = 7 };
+static struct dw_dma_slave puma7_0_tx_param = { .dst_id = 0 };
+static struct dw_dma_slave puma7_0_rx_param = { .src_id = 1 };
+static struct dw_dma_slave puma7_1_tx_param = { .dst_id = 6 };
+static struct dw_dma_slave puma7_1_rx_param = { .src_id = 7 };
 
 static bool lpss_dma_filter(struct dma_chan *chan, void *param)
 {
@@ -99,21 +99,21 @@ static struct pxa_spi_info spi_info_configs[] = {
 		.tx_param = &bsw2_tx_param,
 		.rx_param = &bsw2_rx_param,
 	},
-	[PORT_CGM0] = {
+	[PORT_PUMA7_0] = {
 		.type = LPSS_SSP,
 		.port_id = 0,
 		.num_chipselect = 1,
 		.max_clk_rate = 50000000,
-		.tx_param = &cgm0_tx_param,
-		.rx_param = &cgm0_rx_param,
+		.tx_param = &puma7_0_tx_param,
+		.rx_param = &puma7_0_rx_param,
 	},
-	[PORT_CGM1] = {
+	[PORT_PUMA7_1] = {
 		.type = LPSS_SSP,
 		.port_id = 1,
 		.num_chipselect = 1,
 		.max_clk_rate = 50000000,
-		.tx_param = &cgm1_tx_param,
-		.rx_param = &cgm1_rx_param,
+		.tx_param = &puma7_1_tx_param,
+		.rx_param = &puma7_1_rx_param,
 	},
 };
 
@@ -219,8 +219,8 @@ static const struct pci_device_id pxa2xx_spi_pci_devices[] = {
 	{ PCI_VDEVICE(INTEL, 0x228e), PORT_BSW0 },
 	{ PCI_VDEVICE(INTEL, 0x2290), PORT_BSW1 },
 	{ PCI_VDEVICE(INTEL, 0x22ac), PORT_BSW2 },
-	{ PCI_VDEVICE(INTEL, 0x2b8e), PORT_CGM0 },
-	{ PCI_VDEVICE(INTEL, 0x2b90), PORT_CGM1 },
+	{ PCI_VDEVICE(INTEL, 0x2b8e), PORT_PUMA7_0 },
+	{ PCI_VDEVICE(INTEL, 0x2b90), PORT_PUMA7_1 },
 	{ },
 };
 MODULE_DEVICE_TABLE(pci, pxa2xx_spi_pci_devices);
diff --git a/drivers/spi/spi-pxa2xx.c b/drivers/spi/spi-pxa2xx.c
index 74c4c6e..c524a750 100644
--- a/drivers/spi/spi-pxa2xx.c
+++ b/drivers/spi/spi-pxa2xx.c
@@ -1156,7 +1156,7 @@ static struct pxa2xx_spi_acpi_desc cht_spi_desc = {
 	.cs_num = 2,
 };
 
-static struct pxa2xx_spi_acpi_desc cgm_spi_desc = {
+static struct pxa2xx_spi_acpi_desc puma7_spi_desc = {
 	.cs_num = 2,
 };
 
@@ -1167,8 +1167,7 @@ static struct acpi_device_id pxa2xx_spi_acpi_match[] = {
 	{ "INT3431", 0 },
 	{ "80860F0E", (kernel_ulong_t)&byt_spi_desc },
 	{ "8086228E", (kernel_ulong_t)&cht_spi_desc },
-	{ "8086228E", 0 },
-	{ "80862B8E", (kernel_ulong_t)&cgm_spi_desc },
+	{ "80862B8E", (kernel_ulong_t)&puma7_spi_desc },
 	{ },
 };
 MODULE_DEVICE_TABLE(acpi, pxa2xx_spi_acpi_match);
diff --git a/drivers/usb/dwc3/dwc3-pci.c b/drivers/usb/dwc3/dwc3-pci.c
index 6246b07..bab6bc2 100644
--- a/drivers/usb/dwc3/dwc3-pci.c
+++ b/drivers/usb/dwc3/dwc3-pci.c
@@ -33,7 +33,7 @@
 #define PCI_DEVICE_ID_INTEL_BYT		0x0f37
 #define PCI_DEVICE_ID_INTEL_MRFLD	0x119e
 #define PCI_DEVICE_ID_INTEL_CHT		0x22b7
-#define PCI_DEVICE_ID_INTEL_CGM		0x2bb7
+#define PCI_DEVICE_ID_INTEL_PUMA7	0x2bb7
 
 struct dwc3_pci {
 	struct device		*dev;
@@ -64,7 +64,7 @@ static int dwc3_pci_register_phys(struct dwc3_pci *glue)
 
 	if (pci_dev->vendor == PCI_VENDOR_ID_INTEL &&
 			(pci_dev->device == PCI_DEVICE_ID_INTEL_CHT ||
-             pci_dev->device == PCI_DEVICE_ID_INTEL_CGM))
+			pci_dev->device == PCI_DEVICE_ID_INTEL_PUMA7))
 		pdev = platform_device_alloc("intel-cht-otg", 0);
 	else
 		pdev = platform_device_alloc("usb_phy_gen_xceiv", 0);
@@ -244,7 +244,7 @@ static const struct pci_device_id dwc3_pci_id_table[] = {
 		.driver_data	= (kernel_ulong_t)&dwc3_intel_cht,
 	},
 	{
-		PCI_DEVICE(PCI_VENDOR_ID_INTEL, PCI_DEVICE_ID_INTEL_CGM),
+		PCI_DEVICE(PCI_VENDOR_ID_INTEL, PCI_DEVICE_ID_INTEL_PUMA7),
 		.driver_data	= (kernel_ulong_t)&dwc3_intel_cht,
 	},
 	{  }	/* Terminating Entry */
-- 
2.10.1

